### PR TITLE
Editorial: Introduce OrdinaryGet/Set/OwnPropertyKeys and use them

### PR DESCRIPTION
**Pull request by annevk**

### CODE REVIEW

**Comment by ljharb** [comment]
this looks much cleaner to me 👍

**Comment by bterlson** [comment]
I think this is good. Slightly more noise at the definition of the slots, but much less noise at every other reference.

Two things, and I can pull this in:

1. Remove the es6id for the new clauses. New clauses should not have this metadata (it's supposed to identify clauses that came from the ES6 doc)
2. Similar treatment for all the other ordinary object slots (may not be required now, but could be in the future, and so is good to be consistent I think).

**Comment by annevk** [comment]
Rebased and amended to take care of that. I also found one more usage of ordinary [[Delete]] that I replaced with OrdinaryDelete. (It even had ReturnIfAbrupt!)

**Comment by annevk** [comment]
Oh also, I didn't try to build and I used copy-and-paste. So beware of that. I looked it twice over, but it's easy to miss things.

**Comment by bterlson** [comment]
@annevk thanks so much! I will test this build before I pull it in just to double-check.

File: spec.html [hunk 1]
```diff
@@ -6269,55 +6269,87 @@
     <!-- es6num="9.1.1" -->
     <emu-clause id="sec-ordinary-object-internal-methods-and-internal-slots-getprototypeof">
       <h1>[[GetPrototypeOf]] ( )</h1>
-      <p>When the [[GetPrototypeOf]] internal method of _O_ is called the following steps are taken:</p>
+      <p>When the [[GetPrototypeOf]] internal method of _O_ is called, the following steps are taken:</p>
       <emu-alg>
-        1. Return the value of the [[Prototype]] internal slot of _O_.
+        1. Return OrdinaryGetPrototypeOf(_O_).
```
Comment (bterlson) [inline spec.html]: Add !.

File: spec.html [hunk 2]
```diff
@@ -6269,55 +6269,87 @@
...
-        1. Let _done_ be *false*.
-        1. Repeat while _done_ is *false*,
-          1. If _p_ is *null*, let _done_ be *true*.
-          1. Else, if SameValue(_p_, _O_) is *true*, return *false*.
-          1. Else,
-            1. If the [[GetPrototypeOf]] internal method of _p_ is not the ordinary object internal method defined in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots-getprototypeof"></emu-xref>, let _done_ be *true*.
-            1. Else, let _p_ be the value of _p_'s [[Prototype]] internal slot.
-        1. Set the value of the [[Prototype]] internal slot of _O_ to _V_.
-        1. Return *true*.
+        1. Return OrdinarySetPrototypeOf(_O_, _V_).
```
Comment (bterlson) [inline spec.html]: Add !

File: spec.html [hunk 3]
```diff
@@ -6269,55 +6269,87 @@
...
     </emu-clause>
 
     <!-- es6num="9.1.3" -->
     <emu-clause id="sec-ordinary-object-internal-methods-and-internal-slots-isextensible">
       <h1>[[IsExtensible]] ( )</h1>
-      <p>When the [[IsExtensible]] internal method of _O_ is called the following steps are taken:</p>
+      <p>When the [[IsExtensible]] internal method of _O_ is called, the following steps are taken:</p>
       <emu-alg>
-        1. Return the value of the [[Extensible]] internal slot of _O_.
+        1. Return OrdinaryIsExtensible(_O_).
```
Comment (bterlson) [inline spec.html]: Add !

File: spec.html [hunk 4]
```diff
@@ -6475,60 +6517,84 @@
...
     </emu-clause>
 
     <!-- es6num="9.1.11" -->
     <emu-clause id="sec-ordinary-object-internal-methods-and-internal-slots-enumerate">
       <h1>[[Enumerate]] ()</h1>
-      <p>When the [[Enumerate]] internal method of _O_ is called the following steps are taken:</p>
+      <p>When the [[Enumerate]] internal method of _O_ is called, the following steps are taken:</p>
       <emu-alg>
-        1. Return an Iterator object (<emu-xref href="#sec-iterator-interface"></emu-xref>) whose `next` method iterates over all the String-valued keys of enumerable properties of _O_. The Iterator object must inherit from %IteratorPrototype% (<emu-xref href="#sec-%iteratorprototype%-object"></emu-xref>). The mechanics and order of enumerating the properties is not specified but must conform to the rules specified below.
+        1. Return ? OrdinaryEnumerate(_O_).
```
Comment (bterlson) [inline spec.html]: This entire clause will change once #367 is accepted. Perhaps undo any changes to this clause so rebase is easier. OrdinaryEnumerate is now basically EnumerateObjectProperties in #367 so from a layering perspective should work identically.

File: spec.html [hunk 5]
```diff
@@ -6555,15 +6622,24 @@
...
       <emu-alg>
-        1. Let _keys_ be a new empty List.
-        1. For each own property key _P_ of _O_ that is an integer index, in ascending numeric index order
-          1. Add _P_ as the last element of _keys_.
-        1. For each own property key _P_ of _O_ that is a String but is not an integer index, in ascending chronological order of property creation
-          1. Add _P_ as the last element of _keys_.
-        1. For each own property key _P_ of _O_ that is a Symbol, in ascending chronological order of property creation
-          1. Add _P_ as the last element of _keys_.
-        1. Return _keys_.
+        1. Return OrdinaryOwnPropertyKeys(_O_).
```
Comment (bterlson) [inline spec.html]: Add !

File: spec.html [hunk 6]
```diff
@@ -8094,7 +8170,7 @@
         <p>When the [[OwnPropertyKeys]] internal method of a module namespace exotic object _O_ is called the following steps are taken:</p>
         <emu-alg>
           1. Let _exports_ be a copy of the value of _O_'s [[Exports]] internal slot.
-          1. Let _symbolKeys_ be the result of calling the default ordinary object [[OwnPropertyKeys]] internal method (<emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots-ownpropertykeys"></emu-xref>) on _O_ passing no arguments.
+          1. Let _symbolKeys_ be OrdinaryOwnPropertyKeys(_O_).
```
Comment (bterlson) [inline spec.html]: Add !

**Comment by bterlson** [comment]
Added a few more comments (sorry!)

Checked the document structure and everything is good. Build is fine, output looks great!

**Comment by annevk** [comment]
I also added ! in front of OrdinaryPreventExtensions and OrdinaryGetOwnProperty too. It seems you missed those.

I'm a little concerned about [[Enumerate]] going away, but removed its refactoring and left a comment on the relevant PR.

**Comment by bterlson** [comment]
I wanted to land #367 first but I will pull this in now.

**Comment by bterlson** [comment]
Thanks much @annevk!

**Comment by Skiddish** [comment]
https://github.com/whatwg/html/pull/638

**Pull request closed without merging**